.github/workflows/go.yml

```diff
@@ -394,12 +401,12 @@ jobs:
   on-failure:
     runs-on: ubuntu-latest
     if: failure() && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v'))
-    needs: [go-release, post-release, nix-shell-test, push-docker]
```

Comment:
What's the idea here? I don't think this does anything, since those new jobs that were added are already implied dependencies. I think the problem with `job.status` is that it refers to the *job* `on-failure`, not the other jobs. Is there a `workflow.status`?